internal/controller/operatorconfigmap_controller.go

```diff
@@ -469,6 +476,32 @@ func (c *OperatorConfigMapReconciler) reconcileDelegatedCSI() error {
 	return nil
 }
 
+func (c *OperatorConfigMapReconciler) getDomainLabels() []string {
+	topologyLabels := []string{
+		"kubernetes.io/hostname",
+		"topology.kubernetes.io/zone",
+		"topology.rook.io/rack",
+	}
+
+	// list node labels
+	nodes := &corev1.NodeList{}
+	err := c.List(c.ctx, nodes)
+	if err != nil {
+		c.log.Error(err, "failed to list nodes labels")
+	}
+
+	actualTopologyLabels := []string{}
+	for _, node := range nodes.Items {
+		for _, label := range topologyLabels {
+			if node.GetLabels()[label] != "" {
+				actualTopologyLabels = append(actualTopologyLabels, label)
+			}
+		}
+	}
```

Comment:
May I propose the have a map of the annotation to an counter, then go over the nodes to add to the counter.
This will allow you to ignore any key where the counter (value) is not the same as the number of nodes.

This way the complexity of the solution will be 0(N) and not o(n*m)

Reply:
PS: or it expected to have labels on all the nodes, even master nodes?
Or if we really want this I can count total storage nodes by looking at `cluster.ocs.openshift.io/openshift-storage=`